### ISSUE TITLE
User access to shared workspace not granted if no Astronomer account

### ISSUE THREAD

**Issue by paolaperaza** [astronomer/astronomer#249]
**Describe the bug**
If a user that does _not_ have an existing account on Astronomer is invited to a shared workspace, that user will not automatically be granted access to that workspace when the user creates an account directly from the invitation email.

**To Reproduce**
1. Create an account on Astronomer, or log into an existing account (User A)
2. Invite a user (User B) who does NOT have an account on Astronomer to that workspace
3. User B accepts the invite email
4. User B creates an account on Astronomer, via username + password
5. User B logs in
6. User B does _not_ see User A's workspace listed on their account, nor do they have another form of accepting the invite to said workspace

**Expected behavior**
If a non-user receives an invite to someone else's workspace on Astronomer, they should have automatic access to that workspace upon accepting the invite via email and creating an account, (regardless of sign up authorization method).

**Desktop (please complete the following information):**
 - OS: OS 10.14.2
 - Browser Chrome

**Additional context**
We were not able to test if this happened if a non-user accepts the email invite and creates an account using Google OAuth or GitHub, instead of the username + password route.

**Comment by paolaperaza** [comment]
@schnie Have had 2 prospects hit this bug recently and not be able to add their coworkers to a workspace 😬 

Any chance it'll be fixed in Houston2?

**Comment by schnie** [comment]
@paolaperaza yep, it should be fixed in Houston2. If you want, you can test it out to be sure on alpha.datarouter.ai.

**Comment by paolaperaza** [comment]
@schnie Long overdue but I just tested this again on demo.datarouter.ai and didn't see an issue.

I did the following:

1. Invited user B (personal email address)
2. Clicked on email invite link and created an account on demo.datarouter.ai in incognito mode
3. As User B, immediately see workspace I was invited to by user A
4. As User A, I see user B in my workspace turn from "Pending" to "Joined" as soon as they created an account

I then tested deleting and re-adding. Since they already had an account on the cluster, I did _not_ get an invite but immediately saw that User B "Joined" the Workspace (skipped the "Pending" step).

If that's all expected, @schnie feel free to confirm and close out!
